common/middleware/src/meta_rules.rs

```diff
@@ -0,0 +1,516 @@
+use crate::{Middleware, Status};
+//use log::debug;
+use http::types::body::{KeyValueMap, LineBufferMut};
+use std::collections::HashMap;
+
+/// Env config options
+static LOGDNA_META_APP: &str = "LOGDNA_META_APP";
+static LOGDNA_META_HOST: &str = "LOGDNA_META_HOST";
+static LOGDNA_META_ENV: &str = "LOGDNA_META_ENV";
+static LOGDNA_META_FILE: &str = "LOGDNA_META_FILE";
+static LOGDNA_META_K8S_FILE: &str = "LOGDNA_META_K8S_FILE";
+static LOGDNA_META_JSON: &str = "LOGDNA_META_JSON";
+static LOGDNA_META_ANNOTATIONS: &str = "LOGDNA_META_ANNOTATIONS";
+static LOGDNA_META_LABELS: &str = "LOGDNA_META_LABELS";
+
+//TODO: extract to LogConfig
+pub struct MetaRulesConfig {
+    pub app: Option<String>,
+    pub host: Option<String>,
+    pub env: Option<String>,
+    pub file: Option<String>,
+    pub k8s_file: Option<String>, // for k8s lines, applied after "file"
+    pub meta: Option<String>,
+    pub annotations: Option<String>,
+    pub labels: Option<String>,
+}
+
+impl MetaRulesConfig {
+    pub fn default() -> Self {
```

Comment:
Since it's all None's we could get rid of this function and add the 

`#[derive(Default)]` annotation to the MetaRulesConfig struct and the compiler will add an `impl Default for MetaRulesConfig` for us

Reply:
+1. updating.